src/components/ui/File.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Icon from './Icon';
+import Button from './Button';
+import './File.css';
+
+/*
+*
+*  File components
+*
+* */
+
+const withDownload = Component => props => {
+  const handleDownload = event => {
+    event.preventDefault();
+    const a = document.createElement('a');
+    a.href = props.data;
+    a.download = props.name;
+    a.click()
+  }
+
+  return <Component handleDownload={handleDownload} {...props}/>
+}
+
+const Image = withDownload(({ src, alt, handleDownload, ...rest}) => (
+  <div className="image-file">
+    <div className="overlay" >
+      <Icon iconName="downArrow" onClick={handleDownload} />
+    </div>
+    <img src={src} alt={alt} {...rest} />
+  </div>
+));
+
+const Default = withDownload(({ name, handleDownload, type }) => (
+  <div
+    className="default-file"
+  >
+    <div className="content">
+      <Icon iconName="download"/>
+      <div className="titles">
+       <p>{name}</p><p>{type}</p>
+      </div>
+    </div>
+    <Button onClick={handleDownload} label="download" color="success" />
+  </div>
+));
+
+/*
+*
+* File hub component :
+*
+* */
+
+const File = props => {
+  const { name, data } = props;
+  const type = props['content-type'];
+
+  if (['jpeg', 'png', 'jpg', 'gif', 'heic', 'heif'].find(check => type.indexOf(check) !== -1)) {
```

Comment:
I'd move an array to a constant: `ALLOWED_TYPE`. You could also use `includes` instead of `indexOf` :wink: